docs/08-docker/02-windows-docker-images.mdx

```diff
@@ -17,70 +17,111 @@ found in
+newer with C++ compilers and Windows 10 SDK (recommended) * Visual Studio 2019 with C++ compilers and
+Windows 10 SDK * Visual Studio 2017 with C++ compilers and Windows 10 SDK * Visual Studio 2015 with
+C++ compilers and Windows 10 SDK Visual Studio 2017 (or newer) is detected using `vswhere.exe` as well
+as VSCOMNTOOLS environment variables. Visual Studio 2015 is detected by looking at "SOFTWARE\Microsoft\VisualStudio\14.0_Config\InstallDir"
+in the registry as well as VSCOMNTOOLS environment variables. Windows 10 SDK is detected by looking
+at "SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0\InstallationFolder" in the registry.
+Unable to detect any compatible Visual Studio installation! * Found Visual Studio 2022 installation
+without C++ tool components Windows 10 SDK is not installed. You can install from here: https://developer.microsoft.com/en-us/windows/downloads/windows-10-sdk/
+```
 
-_Add to the system registry (in a container) path and version to Windows SDK_
+</details>
 
-```powershell
-$windowsSdkDirectory = "HKLM:\SOFTWARE\Wow6432Node\Microsoft\Microsoft SDKs\Windows\v10.0"
+There are two options to avoid this obstacle:
+
+1. (_Recommended_) Inject/mount Microsoft Visual Studio with VC++ and Windows 10 SDK from the host
+   system into the container during runtime.
+2. Build a custom image and push it to a docker registry.
 
-New-Item -Path $windowsSdkDirectory -Force | Out-Null
+#### Option 1. Injecting VC++ from the host system into the container
 
-New-ItemProperty -Path $windowsSdkDirectory `
-     -Name InstallationFolder `
-     -PropertyType String `
-     -Force `
-     -Value "C:\Program Files (x86)\Windows Kits\10\" | Out-Null
+> These instructions have been validated with Visual Studio 2022 and Unity 2021 LTS.
 
-New-ItemProperty -Path $windowsSdkDirectory `
-     -Name ProductVersion `
-     -PropertyType String `
-     -Force `
-     -Value $env:WINDOWS_SDK_VERSION | Out-Null
+_Install Visual Studio Build Tools_
+
+The recommended way to install the correct dependencies on the host is with
+[Chocolatey](https://chocolatey.org/). This works with Windows 10/11, Windows Server, and Windows
+Server Core.
+
+```powershell
+choco install visualstudio2022-workload-vctools --no-progress -y
 ```
 
-_Mount list of volumes_
+There is a way to install the same dependencies from the graphical
+[Visual Studio Installer](https://visualstudio.microsoft.com/downloads/), but the workload component
+options in the installer frequently change.
+
+_The following directories need to be bind-mounted into the container_
 
 - C:\Program Files (x86)\Windows Kits
 - C:\Program Files (x86)\Microsoft Visual Studio
 - C:\Program Files\Microsoft Visual Studio
 - C:\ProgramData\Microsoft\VisualStudio
 
+> Please confirm all of the above directories exist in your installation. Some may be missing and
+> things might still work OK without them. Omit the missing directories in the following steps.
```

Comment:
You might want to say pick between the two Microsoft visual studio options as missing the windows kits and program data folders will cause the build to fail. Maybe have a section for windows kits and program data then a second section that says pick one of the following based on your installation

Reply:
Fair enough.